src/core/geometry/qgscurve.cpp

```diff
@@ -286,7 +286,7 @@ double QgsCurve::sinuosity() const
 Qgis::AngularDirection QgsCurve::orientation() const
 {
   double a = 0;
-  sumUpArea( a );
+  sumUpArea( a, true );
```

Comment:
Yep, the real issue is in QgsLineString::reversed

Reply:
Indeed, I had stopped here because it worked, forgetting the simplest point that I was testing the reversed function. Thanks, @nyalldawson, for opening my eyes.

Reply:
Thanks @lbartoletti ! This looks good nkw